crabTask.py

```diff
@@ -383,11 +383,26 @@ def findOutputFile(self, taskOutput, jobId):
     outputName, outputExt = os.path.splitext(self.getCrabJobOutput())
     fileName = f'{outputName}_{jobId}{outputExt}'
     outputFiles = []
-    for root, dirs, files in os.walk(taskOutput):
-      for file in files:
-        if file == fileName:
-          filePath = os.path.join(root, file)
-          outputFiles.append(filePath)
+    if taskOutput.startswith("/"):
+      for root, dirs, files in os.walk(taskOutput):
+        for file in files:
+          if file == fileName:
+            filePath = os.path.join(root, file)
+            outputFiles.append(filePath)
+    else:
+      def RecursiveGfal(path, fileName):
+        _, output, _ = sh_call(['gfal-ls', '-t 7200', path,],
+                               shell=False, env={'X509_USER_PROXY': get_voms_proxy_info()['path']}, catch_stdout=True, verbose=0)
+        output = [o for o in output.strip().split("\n")]
+        outputFiles = []
+        for o in output:
+          if not o.endswith(".tar"):
+            outputFiles.extend(RecursiveGfal(os.path.join(path, o), fileName))
+          else:
+            if o == fileName:
+              outputFiles.append(os.path.join(path, o))
+        return outputFiles
```

Comment:
I suggest moving this function to sh_utils.py and renaming it to `gfal_ls_recursive` to match naming convention